### PR TITLE
Add SpringSecurity

### LINKED ISSUE #35: Implement OAuth with GitHub
As a user of the application
I want to be able to log in to the application using my GitHub credentials
So that I can easily access the application without creating a new account and password, leveraging my existing GitHub account for authentication.

# Acceptance Criteria:

1. **OAuth Integration:**
   - The application should integrate with GitHub's OAuth service.
   - Users should be able to initiate the login process by clicking a "Log in with GitHub" button.

2. **Redirect to GitHub:**
   - When the "Log in with GitHub" button is clicked, users should be redirected to GitHub's authorization page.

3. **Authorization Code:**
   - After successful authentication on GitHub, users should be redirected back to the application with an authorization code.

4. **Access Token Exchange:**
   - The application should exchange the authorization code for an access token with GitHub.

5. **User Information Retrieval:**
   - Using the access token, the application should retrieve the user's basic profile information from GitHub (e.g., username, email).

6. **User Session:**
   - Upon successful retrieval of user information, the application should create a user session and log the user in.

7. **Error Handling:**
   - The application should handle errors gracefully during the OAuth process (e.g., failed authorization, network issues) and provide appropriate feedback to the user.

8. **Security:**
   - Ensure the OAuth implementation follows best security practices, including secure storage of tokens and protection against common vulnerabilities.

9. **Documentation:**
   - Provide clear documentation for the OAuth implementation, including steps for configuring GitHub OAuth and troubleshooting common issues.

### PR DESCRIPTION
**Pull request by jonashonecker**
- backend: Add SpringSecurity configuration and endpoint to retrieve user information
- frontend: Add login button

### CODE REVIEW

File: backend/src/main/java/com/neuefische/team2/backend/auth/domain/AuthDTO.java
Comment (josch87) [inline backend/src/main/java/com/neuefische/team2/backend/auth/domain/AuthDTO.java:1]: It is not a DTO as you do not transfer anything. It is a representation of the githubUserProfile. So you could name it something like GitHubUserProfile.

File: frontend/src/components/ButtonLogin/ButtonLogin.styled.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { FaCircleUser } from "react-icons/fa6";
+
+export const StyledFaCircleUser = styled(FaCircleUser)`
+  width: 30px;
+  height: 30px;
+
+  &:hover {
+    cursor: pointer;
+    transform: rotate(360deg);
+    transition: transform 0.5s ease-in-out;
```
Comment (josch87) [inline frontend/src/components/ButtonLogin/ButtonLogin.styled.ts:11]: Very nice!

File: frontend/src/components/ButtonLogin/ButtonLogin.styled.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  }
+
+  &:active {
+    transform: scale(0.9);
+    transition: transform 0.2s;
```
Comment (josch87) [inline frontend/src/components/ButtonLogin/ButtonLogin.styled.ts:16]: Nice!

File: frontend/src/components/ButtonLogin/ButtonLogin.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { StyledFaCircleUser } from "./ButtonLogin.styled.ts";
+
+type ButtonLogin = {
```
Comment (josch87) [inline frontend/src/components/ButtonLogin/ButtonLogin.tsx:3]: Please use "ButtonLoginProps" as name

**Review by josch87 — CHANGES_REQUESTED** [review]
1. Home page is showing an error instead of saying you should log in first in order to create your first restaurant. 
2. When logged in it is currently possible to login again. Also it is not displayed anywhere, that you are logged in. 
3. Backend tests are not working anymore (but we have a look into that at todays session)

File: backend/src/main/java/com/neuefische/team2/backend/auth/domain/AuthDTO.java
Comment (jonashonecker) [reply backend/src/main/java/com/neuefische/team2/backend/auth/domain/AuthDTO.java:1]: Done!

**Review by josch87 — APPROVED** [review]

**Pull request merged**